project/apps/matching-service/src/matchExpiry/matchExpiry.service.ts

```diff
@@ -1,11 +1,29 @@
-import { Injectable } from '@nestjs/common';
+import { Injectable, Logger } from '@nestjs/common';
 import { MatchRedis } from 'src/db/match.redis';
+import { MatchingGateway } from 'src/matching.gateway';
 
 @Injectable()
 export class MatchExpiryService {
-  constructor(private readonly matchRedis: MatchRedis) {}
+  private readonly logger = new Logger(MatchExpiryService.name);
+  constructor(
+    private readonly matchRedis: MatchRedis,
+    private readonly matchGateway: MatchingGateway,
+  ) {}
 
-  handleExpiryMessage(_expiryMessage: any) {
+  async handleExpiryMessage(expiredMatchId: string) {
     // Perform some sort of message handling
+    const matchRequest =
+      await this.matchRedis.removeMatchRequest(expiredMatchId);
+    if (!matchRequest) {
+      this.logger.warn(
+        `Match request with id ${expiredMatchId} does not exist`,
+      );
+      return;
```

Comment:
nit: should this be a warn? cause it isn't really an error/warning case but an expected part of our flow.

Example:
1. User A sends match request but no matches found yet, so expiry gets enqueued
2. User B sends match request gets User A as matched
3. User's A match expiry gets consumed and finds User A is not there anymore

Reply:
yup, perhaps debug would be more appropriate in this case, will make the changes

Regarding redis insights, once docker has been deployed, you can head to localhost:5540 to access the GUI. At the main page, you would have to add redis as url followed by the corresponding port the db is hosted on. Thereon, you can use it for monitoring of DB changes, CRUD and etc easily

Helpful resource:
https://redis.io/docs/latest/develop/connect/insight/